### DISCUSSION TITLE
榕典新数据模型

### DISCUSSION THREAD

**Discussion by ztl8702** [MindongLab/yngdieng#692] [产品功能 ideas]
近期要设计一下。
先开个issue记录问题。

**Comment by ztl8702** [comment]
## 读音类型的问题

zingzeu_words 中设想的 SC（读音类型）包括 ：连读音、本字音两种。V包括：市区、长乐、etc。SC和V是两个维度，可以任意组合。

* 地名读音往往不适用于“连读音/本字音”这样简单划分
   * @Guanchishan  地名一定名从主人，就好像普通话里也把六安这个地名读成lu而非liu一样。他die-la这个地名基本还是遵守福州市区的音调体系的

**Comment by ztl8702** [comment]
## 不同口音的处理

* 是“音频本位”：音频拥有对应的读音。还是：词条拥有读音，读音拥有音频？
#560 
   * @guanchishan：感觉后者有利于数据的整理和呈现，更有条理：先分读音，再呈现同一读音的不同音频

**Comment by Guanchishan** [comment]
## 同义词、近义词

目前是在contrib里面开了个同义词字段记录同义词。但是这会担心由于人工维护导致数据不一致的问题。
应该中心化管理同义词信息。用一个意义素统领诸多同义词。可能要另外开张表来记录这些信息。

@ztl8702 这个其实是一个网络模型（图 / linked-data）。看一下 WordNet，Wikitionary，Wikidata等怎么处理。
* @guanchishan：wikitionary的思路是，搞一个同一的template页面记录这些信息，然后在不同词条调用同一个template。Wikidata似乎是在具体的词条页手动添加信息，但是会检查数据一致性，提示你需要手动为对应条目也添加信息。wikidata这么设计的逻辑应该是，数据处理是机器人的事情，机器人可以轻松的去两边添数据，这和我们人手动去两边翻数据添的繁琐程度的是不一样的。wikitionary的思路应该对人类更友好。

**Comment by Guanchishan** [comment]
## 待遇表現
日语词。中文不知道怎么叫。意思是在社交活动所涉及的语言主体（说话人，写作者）与对方、话题人物三者之间的上下、亲疏等人际关系，以及这些关系在语言中的反映。体现在福州话里，就好像我们不敢在长辈面前说“講脬”“𡳞脬”一样，在吃宴席的时候不好说“食飽了”，因为“BA”一声不好听，只好说“食完了”，甚至听过福清牧师讲道说“今天”不说“今旦”。这也需要另外一张表单去按庄重程度线性排列（目前福州只想到庄重程度一个维度，怕搞太复杂；日语的话一般是坐标轴：横轴是庄重，纵轴是口语、书面）

**Comment by ztl8702** [comment]
## 何为条目，何为义项

zingzeu_words 目前理论上是给每个（字，音，义）组合 开一个ID。同字不同音-> 不同ID；同字同音不同义-> 不同ID（虽然说不同义，一般来说音也会不同？ e.g. 树兜）。但是每个ID又可能指向一个冯版条目、一个李版条目。冯、李对词义的划分，可能有所不同。

**Comment by ztl8702** [comment]
## 短语、熟语、歌谣、诗歌能不能加进来？

有的冯条目就是短语或熟语。未来也计划把几本熟语数据导入。熟语也用zingzeu ID？

* @guanchishan 我赞成。以前想法见：https://github.com/MindongLab/foochow-idioms/issues/96 “原本构思出了三种语词字段：word（字词）、sentence（句，各种俗语与词典提供的各种例句都算在内）、work（如《圣经》《[筅堂洗厝好过年](https://github.com/MindongLab/cdo-corpus/blob/master/essay/Spring%20Festival%20in%20Fuzhou%20by%20Fujian%20Provincial%20Library/%E7%AD%85%E5%A0%82%E6%B4%97%E5%8E%9D%E5%A5%BD%E8%BF%87%E5%B9%B4.tsv)》或者[闽剧剧本](https://github.com/MindongLab/cdo-corpus/blob/master/opera/Min%20opera/%E6%98%A5%E8%8D%89%E9%97%96%E5%A0%82%C2%B7%E6%89%9B%E8%BD%8E%E7%89%87%E6%AE%B5%5Bcdo-latn%5D.srt)）。字词组成句，句组成文章。反过来，文章可析成一句句话去点击查看详情，[句到字词亦然](https://github.com/MindongLab/foochow-idioms/issues/42)。（正如当时学长提出[歌谣与俗语宜分属不同数据结构](https://github.com/MindongLab/foochow-idioms/issues/57#issue-278785842)。）后来觉得正如以前对词汇、例句、俗语（#56、#64）等定义与分野的探讨一样，他们实在如光谱一般，是过渡的概念，彼此之间界限并不分明。不如都抹平，直接视为word，word之间再彼此调用作为用例等等？word本身可以通过声明体裁tag明确其为歌谣、歇后语乃至戏文，还可以加单字tag以明确这是单字，加新闻稿tag明确这是新闻稿等等。”
 * @guanchishan 不过有的俗语过长，在现有的榕典词汇界面不一定好显示。不然我们定个粗暴点的标准：产生分句则视为idiom，产生韵脚、复沓、起兴等修辞手法则为歌谣，内容不超过一个分句的和榕典其他词汇、成语等视为同物？

**Comment by Guanchishan** [comment]
> 短语、熟语、歌谣、诗歌能不能加进来？
> 有的冯条目就是短语或熟语。未来也计划把几本熟语数据导入。熟语也用zingzeu ID？

不过在实践过程中，发现除了使用zingzeu ID的熟语本身，还需要两种ID：
一是**异文**。比如由于传唱、传抄等导致训读字本字混用、混说，以及衍文等问题，从而产生同一熟语的不同版本。（如[月光光照池塘](https://github.com/MindongLab/foochow-idioms-data/commit/fe47230abdceb38313f2ebe89c85f40793a900e1)。）这些不同版本本身肯定可以从属于不同zingzeuID，但是也要俾使从属于同一异文树下。
一是**同题**。比如真鸟囝不仅可以啄菠菠（三岁孩儿会唱歌那首），还能食番柿（飞机大炮着准备，解放台湾真容易），还能[抗病毒](https://mp.weixin.qq.com/s?__biz=MzA4MDIxMzU2OA==&mid=2660817388&idx=1&sn=37bfa0fe77eacd2b4c41f653b6bbf23d&chksm=84cbdac3b3bc53d57076404979af572bd9deb18820fc95dbbc7127d3ed26a543c7d84eea106b&token=662268260&lang=zh_CN#rd)，等等。很显然这么多真鸟囝，已经不只是异文的关系了，根本就是不同作品。但却属于同一个题目，可以设置一个“异文”之上的同题ID？

这些ID关系如下：
![image](https://user-images.githubusercontent.com/30859354/97547342-a5e15280-1a08-11eb-846c-ecb55f90ba3c.png)
